### PR TITLE
Fixes review issue 25498

### PR DESCRIPTION
**Pull request by mykrupp**
https://github.com/project-chip/connectedhomeip/issues/25498

### CODE REVIEW

File: examples/lock-app/silabs/efr32/src/LockManager.cpp [hunk 1]
```diff
@@ -682,15 +687,15 @@ bool LockManager::setLockState(chip::EndpointId endpointId, DlLockState lockStat
     }
 
     // Check the PIN code
-    for (uint8_t i = 0; i < kMaxCredentialsPerUser; i++)
+    for (auto currentCredential : mLockCredentials[to_underlying(CredentialTypeEnum::kPin)])
```
Comment (bzbarsky-apple) [inline examples/lock-app/silabs/efr32/src/LockManager.cpp:690]:
```suggestion
    for (const auto & currentCredential : mLockCredentials[to_underlying(CredentialTypeEnum::kPin)])
```

**Review by bzbarsky-apple — CHANGES_REQUESTED** [review]
This does not look like it addresses https://github.com/project-chip/connectedhomeip/pull/25465#discussion_r1127001709 or https://github.com/project-chip/connectedhomeip/pull/25465#discussion_r1127004425 in the SiWx918 LockManager.cpp

File: examples/lock-app/silabs/SiWx917/src/LockManager.cpp [hunk 2]
```diff
@@ -682,15 +689,15 @@ bool LockManager::setLockState(chip::EndpointId endpointId, DlLockState lockStat
     }
 
     // Check the PIN code
-    for (uint8_t i = 0; i < kMaxCredentialsPerUser; i++)
+    for (const auto currentCredential & : mLockCredentials[to_underlying(CredentialTypeEnum::kPin)])
```
Comment (bzbarsky-apple) [inline examples/lock-app/silabs/SiWx917/src/LockManager.cpp:692]:
```suggestion
    for (const auto & currentCredential : mLockCredentials[to_underlying(CredentialTypeEnum::kPin)])
```

File: examples/lock-app/silabs/efr32/src/LockManager.cpp [hunk 3]
```diff
@@ -682,15 +687,15 @@ bool LockManager::setLockState(chip::EndpointId endpointId, DlLockState lockStat
     }
 
     // Check the PIN code
-    for (uint8_t i = 0; i < kMaxCredentialsPerUser; i++)
+    for (const auto currentCredential & : mLockCredentials[to_underlying(CredentialTypeEnum::kPin)])
```
Comment (bzbarsky-apple) [inline examples/lock-app/silabs/efr32/src/LockManager.cpp:690]:
```suggestion
    for (const auto & currentCredential : mLockCredentials[to_underlying(CredentialTypeEnum::kPin)])
```

**Review by bzbarsky-apple — APPROVED** [review]

**Review by yufengwangca — APPROVED** [review]

**Review by selissia — APPROVED** [review]

**Pull request merged**